src/service/filtering/filtering_processing.py

```diff
@@ -0,0 +1,168 @@
+    load_ver - The load version.
+    load_ver_override - Whether or not to override the load version.
+    data_product - The data product id.
+    columns - list of columnar_attribs_common_models.AttributesColumn objects to be filtered on.
+    view_name - The name of the view to use for filtering.
+    count - Whether or not to return the count of matching documents.
+    sort_on - The name of the field to sort on.
+    sort_desc - Whether or not to sort in descending order.
+    filter_conjunction - Whether or not to use a conjunction for filtering.
+    match_spec - A subset specification for matching.
+    selection_spec - A subset specification for selection.
+    keep - A dictionary mapping column names to sets of column types to keep.
+    keep_filter_nulls - Whether or not to keep null values when filtering.
+    skip - The number of documents to skip.
+    limit - The maximum number of documents to return.
+    start_after - The value of the field to start after.
+    trans_field_func - A function to transform the field name to valid column name in the filter query
+    """
+    appstate = app_state.get_app_state(r)
+    filter_query = _get_filter_map(r)
+    if filter_query:
+        if load_ver_override:
+            # If we need this feature than the admin needs to supply the view name to use
+            # via the API
+            raise ValueError("Filtering is not supported with a load version override.")
+
+        if view_name:
+            if not await appstate.arangostorage.has_search_view(view_name):
+                raise ValueError(f"View {view_name} does not exist for collection {coll_id}")
+        else:
+            raise ValueError(f"No search view name configured for collection {coll_id}, "
+                             + f"data product {data_product}. Cannot perform filtering operation")
+
+    columns = {c.key: c for c in columns}
+
+    if sort_on and sort_on not in columns:
+        raise errors.IllegalParameterError(
+            f"No such field for collection {coll_id} load version {load_ver}: {sort_on}")
+    if keep:
+        for col in keep:
+            if col not in columns:
+                raise errors.IllegalParameterError(
+                    f"No such field for collection {coll_id} load version {load_ver}: {col}")
+            if keep[col] and columns[col].type not in keep[col]:
+                raise errors.IllegalParameterError(
+                    f"Column {col} is type '{columns[col].type}', which is not one of the "
+                    + f"acceptable types for this operation: {[t.value for t in keep[col]]}")
+    fs = FilterSet(
+        coll_id,
+        load_ver,
+        collection=arango_coll,
+        view=view_name,
+        count=count,
+        sort_on=sort_on,
+        sort_descending=sort_desc,
+        conjunction=filter_conjunction,
+        match_spec=match_spec,
+        selection_spec=selection_spec,
+        keep=list(keep.keys()) if keep else None,
```

Comment:
Should these not also be translated by the field translation function?

Reply:
`keep` is not used by heatmap. you think we should apply translation func just in case?

Reply:
yeah, if fields are getting translated, they should be translated everywhere

Reply:
Let's not set up the next person to use this code to insert a bug IOW

Reply:
hmmm on a second thought, the intended use of the trans_field_func is specifically for transforming keys within the filter_query. Keys in keep should be exactly what's in the columns.

Reply:
But the user may also be specifying the keys in keep (which they are for genome attribs) in the same way they're specifying the filter fields and so those fields will also need transformation.

Reply:
👍

Reply:
I don't think you need the map at the point you do the translation anymore, just the list right?

Reply:
👍

Reply:
:+1: